chef/cookbooks/hyperv/recipes/windows_features.rb

```diff
@@ -1,34 +1,27 @@
 raise if not node[:platform] == 'windows'
 
-include_recipe 'windows::reboot_handler'
 node.default[:windows][:allow_pending_reboots] = false
 
-if node[:target_platform] =~ /^hyperv/
-  unless node[:windows_features_installed]
-    node.set[:windows_features_installed] = true
-    node.save
-  end
-end
+if node[:target_platform] !~ /^hyperv/
+  node[:features_list][:windows].each do |feature_name, feature_attrs|
+    next if node[:windows_features_installed].include? feature_name
 
```

Comment:
We are testing for array content at a time the array does not yet exist. '''node[:windows_features_installed]''' needs to be initialized in the attributes file first.